src/SchedulerImpl.cpp

```diff
@@ -59,7 +59,10 @@ void SchedulerImpl::Stop() {
   assert(running_.load(std::memory_order_acquire));
   if(!running_.load(std::memory_order_acquire))
     return;
-  keep_running_.store(false, std::memory_order_release);
+  {
+    std::unique_lock<AdaptiveSpinLock> lock(global_mutex_);
```

Comment:
I think, this is not needed. Because we are not checking this value or doing anything with it, a simple atomic operation will go through according to this memory order. So no extra synchronization is needed.